x-pack/filebeat/input/httpjson/input.go

```diff
@@ -112,6 +112,10 @@ func run(
 
 	stdCtx := ctxtool.FromCanceller(ctx.Cancelation)
 
+	if config.Request.Tracer != nil {
+		config.Request.Tracer.Filename = strings.ReplaceAll(config.Request.Tracer.Filename, "*", ctx.ID)
```

Comment:
I think using `*` as a placeholder is not clear. Can we switch this to some named placeholder like `[[id]]`. An asterisk is so commonly used in glob patterns that this is likely to looking like a configuration mistake to experience users.

Reply:
This was my recommendation based on similarity with https://pkg.go.dev/os#MkdirTemp. My argument against `[[id]]` if we are not using templates (an alternative proposal that I made (below) — but argued against due to the additional complexity) is that there is suggestive (but misleading) semantics in something that looks like a template but is not, self-quoting "you put `[[id]]` somewhere and someone will get upset that they can’t use something else. So reduction of choice (or appearance of choice) is a win."

```diff
diff --git a/x-pack/filebeat/input/httpjson/input.go b/x-pack/filebeat/input/httpjson/input.go
index 19d454af80..a55c2bc527 100644
--- a/x-pack/filebeat/input/httpjson/input.go
+++ b/x-pack/filebeat/input/httpjson/input.go
@@ -11,6 +11,8 @@ import (
        "net"
        "net/http"
        "net/url"
+       "strings"
+       "text/template"
        "time"
 
        retryablehttp "github.com/hashicorp/go-retryablehttp"
@@ -110,6 +112,21 @@ func run(
 
        stdCtx := ctxtool.FromCanceller(ctx.Cancelation)
 
+       // "{{" is not a sensible character sequence in a filename,
+       // so if it exists we will assume that the filename is a template.
+       if config.Request.Tracer != nil && strings.Contains(config.Request.Tracer.Filename, "{{") {
+               tmpl, err := template.New("tracer").Parse(config.Request.Tracer.Filename)
+               if err != nil {
+                       return err
+               }
+               var buf strings.Builder
+               err = tmpl.Execute(&buf, ctx.ID)
+               if err != nil {
+                       return err
+               }
+               config.Request.Tracer.Filename = buf.String()
+       }
+
        httpClient, err := newHTTPClient(stdCtx, config, log)
        if err != nil {
                return err
```

Reply:
I had not seen the prior art from `os.MkdirTemp`. So it's good to know there is some precedence even if it looks unfamiliar to me. Let's leave it as is.